### PR TITLE
UNIADAPTIVE Work package (s2u-22x)

### PR DESCRIPTION
**Pull request by stetsche**
https://sakaiproject.atlassian.net/browse/SAK-49276
https://sakaiproject.atlassian.net/browse/SAK-49337
https://sakaiproject.atlassian.net/browse/SAK-49258

### CODE REVIEW

**Comment by mpellicer** [comment]
Question, how many of these tickets depend on S2U-35? How many of them are bugs that affect 22.x and should be merged via the usual process?

File: webapi/src/main/java/org/sakaiproject/webapi/controllers/ConditionController.java [hunk 1]
```diff
@@ -104,6 +106,30 @@ public ResponseEntity<Condition> createCondition(@PathVariable String siteId, @R
...
+                return ResponseEntity.badRequest().build();
+            }
+        }
+        log.info("conds {}", conditions);
+
+        Set<Condition> savedConditions = conditions.stream()
+                .map(conditionService::saveCondition)
+                .collect(Collectors.toSet());
+
+        log.info("savedConds {}", savedConditions);
```
Comment (mpellicer) [inline webapi/src/main/java/org/sakaiproject/webapi/controllers/ConditionController.java:129]: Change any log.info to log.debugs please

**Review by mpellicer — CHANGES_REQUESTED** [review]
Put some comments in the lines that are producing unexpected results and change some log infos to debugs.

Great job @stetsche , awesome piece of work.

File: webapi/src/main/java/org/sakaiproject/webapi/controllers/SiteEntityController.java [hunk 2]
```diff
@@ -0,0 +1,513 @@
...
+                    break;
+                case RESOURCE:
+                    ContentResourceEdit resourceEdit = (ContentResourceEdit) toolEntities.get(entityKey(patchEntity));
+                    Optional<Set<String>> optGroupRefs = Optional.ofNullable(patchEntity.getGroupRefs());
+                    Instant resourceOpenDate = patchEntity.getOpenDate();
+                    Instant resourceCloseDate = patchEntity.getCloseDate();
+                    boolean updateAvailability = ObjectUtils.allNotNull(resourceOpenDate, resourceCloseDate);
+
+                    if (updateAvailability) {
+                        resourceEdit.setAvailabilityInstant(false, resourceOpenDate, resourceCloseDate);
```
Comment (stetsche) [inline webapi/src/main/java/org/sakaiproject/webapi/controllers/SiteEntityController.java:402]: TODO: This seems to not work right for some resource types, as the UI does not show the availability correctly

File: webapi/src/main/java/org/sakaiproject/webapi/controllers/SiteEntityController.java [hunk 3]
```diff
@@ -0,0 +1,513 @@
...
+                    updatedEntities.add(SiteEntityRestBean.of(updatedResource));
+                    break;
+                case FORUM:
+                    DiscussionForum forum = (DiscussionForum) toolEntities.get(entityKey(patchEntity));
+
+                    Optional.ofNullable(patchEntity.getOpenDate())
+                            .ifPresent(openDate -> forum.setOpenDate(Date.from(openDate)));
+
+                    Optional.ofNullable(patchEntity.getCloseDate())
+                            .ifPresent(closeDate -> forum.setCloseDate(Date.from(closeDate)));
```
Comment (stetsche) [inline webapi/src/main/java/org/sakaiproject/webapi/controllers/SiteEntityController.java:453]: TODO: Dates do update here, but at least the two availability boolean need to be set. It' might also be necessary to create a new forum, as I observed that when setting it the dates through the UI.

File: webapi/src/main/java/org/sakaiproject/webapi/controllers/SiteEntityController.java (hunk 2 above)
Comment (stetsche) [inline webapi/src/main/java/org/sakaiproject/webapi/controllers/SiteEntityController.java:402]:
```suggestion
                        // TODO: This seems to not work right for some resource types, as the UI does not show the availability correctly
                        resourceEdit.setAvailabilityInstant(false, resourceOpenDate, resourceCloseDate);
```

File: webapi/src/main/java/org/sakaiproject/webapi/controllers/SiteEntityController.java [hunk 4]
```diff
@@ -0,0 +1,514 @@
...
+                        log.error("Pervious check was not sufficient: {} {}", e.toString(),
+                                ExceptionUtils.getStackTrace(e));
+                        return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
+                    }
+                    updatedEntities.add(SiteEntityRestBean.of(updatedResource));
+                    break;
+                case FORUM:
+                    DiscussionForum forum = (DiscussionForum) toolEntities.get(entityKey(patchEntity));
+
+                    Optional.ofNullable(patchEntity.getOpenDate())
```
Comment (stetsche) [inline webapi/src/main/java/org/sakaiproject/webapi/controllers/SiteEntityController.java:450]:
```suggestion
                    // TODO: Dates do update here, but at least the two availability boolean need to be set. It' might also be necessary to create a new forum, as previously observed when setting it the dates through the UI.

                    Optional.ofNullable(patchEntity.getOpenDate())
```

**Comment by mpellicer** [comment]
Thanks @stetsche terrific work!

**Pull request merged**